src/libstd/collections/hash/set.rs

```diff
@@ -757,6 +757,16 @@ impl<T, S> Eq for HashSet<T, S>
 {
 }
 
+#[unstable(feature = "hashmap_hash", issue = "0")]
+impl<T, S> Hash for HashSet<T, S>
+    where T: Eq + Hash,
+          S: BuildHasher
+{
+    fn hash<H: Hasher>(&self, hasher: &mut H) {
+        self.map.hash(hasher);
```

Comment:
can't `HashSet` could `derive` `Hash` instead of writing an impl by hand?

Reply:
Possibly, I thought of that... but I don't think it derive would add S: BuildHasher` - so I went with manual impl instead.